modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,506 @@
+     */
+    protected TenantStatus getTenantStatus(String tenantDomain) throws LazyLoadingTestException {
+
+        TenantStatus tenantStatus;
+        String requestUrl = webAppURL + "/" + IS_TENANT_LOADED_METHOD_URL + "/" + tenantDomain;
+        try {
+            HttpResponse response = HttpURLConnectionClient.sendGetRequest(requestUrl, null);
+            JSONObject tenantStatusJSON = new JSONObject(response.getData());
+            tenantStatus = new TenantStatus(tenantStatusJSON.getJSONObject("TenantStatus").
+                    getBoolean("tenantContextLoaded"));
+        } catch (IOException ioException) {
+            String customErrorMessage = "IOException when sending the Get request to:" + requestUrl + "\n" +
+                    ioException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, ioException);
+        } catch (JSONException jsonException) {
+            String customErrorMessage = "JSONException when retrieving the values from json object TenantStatus" +
+                    requestUrl + "\n" + jsonException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, jsonException);
+        }
+
+        return tenantStatus;
+    }
+
+    /**
+     * This method will do a REST  GET call to the tenant-info-service (deployed by init() method) to get the web-app
+     * status information.
+     * IT will return the  the deployment status of given web application of given tenant is loaded in Ghost form.
+     *
+     * @param tenantDomain Domain Name of the tenant.
+     * @param webAppName   Name of the Web Application
+     * @return WebAppStatus  which include Tenant status, whether the web-app is started or not, whether the web-app is in
+     * ghost mode or not.
+     * @throws LazyLoadingTestException
+     */
+    protected WebAppStatus getWebAppStatus(String tenantDomain, String webAppName) throws LazyLoadingTestException {
+        String requestUrl = webAppURL + "/" + IS_WEB_APP_LOADED_METHOD_URL + "/" + tenantDomain + "/" + webAppName;
+        WebAppStatus webAppStatus;
+        try {
+
+            HttpResponse response = HttpURLConnectionClient.sendGetRequest(requestUrl, null);
+            JSONObject webAppStatusJSON = new JSONObject(response.getData()).getJSONObject("WebAppStatus");
+            boolean isTenantLoaded = webAppStatusJSON.getJSONObject("tenantStatus").getBoolean("tenantContextLoaded");
+            boolean isWebAppStarted = webAppStatusJSON.getBoolean("webAppStarted");
+            boolean isWebAppGhost = webAppStatusJSON.getBoolean("webAppGhost");
+            webAppStatus = new WebAppStatus(new TenantStatus(isTenantLoaded), isWebAppStarted, isWebAppGhost);
+        } catch (IOException ioException) {
+            String customErrorMessage = "IOException when sending the Get request to:" + requestUrl + "\n" +
```

Comment:
IMO you don't need to append exception message to you log again. Just adding the exception object would be enough